### ISSUE TITLE
`--watch` in docker cmd terminates immediately

### ISSUE THREAD

**Issue by Earlopain** [evanw/esbuild#1848]
I'm trying to set up esbuild in a docker container which uses `--watch` to continuously detect changes. The problem is that the process seems to terminate immediately without  building anything. I'm not  seeing any extra output with `--log-level=verbose` and the exit code is 0. Running the command inside the container manually with `docker-compose run repro yarn build` works fine though. Removing the `--watch` parameter from `package.json` also makes it work, it seems to just have problems with `--watch`.

Here's  the relevant output from `docker-compose up`:
```
esbuild-repro-1  | yarn run v1.22.17
esbuild-repro-1  | $ esbuild javascript/test.js --outdir=/app/build --log-level=verbose --watch
esbuild-repro-1  | Done in 0.09s.
esbuild-repro-1 exited with code 0
```

I have also made a minimal reproducable example you can take a look at here https://github.com/Earlopain/esbuild-repro. Do note that the base is alpine but I got the same behaviour with a  ubuntu base. `node_modules` is used exclusivly from inside the container because of the volume, nothing  that might be installed locally should interfere.

**Comment by evanw** [comment]
I believe this is because esbuild ends watch mode if stdin is closed, and Docker is closing stdin. You could try having stdin not be closed to get this to work. I have no idea how to use Docker so I don't know what option to use for that, but doing something like using `sleep 9999999999 | esbuild` instead of just `esbuild` seems to work.

**Comment by Earlopain** [comment]
I can confirm that using it with `sleep` does indeed work. I don't know if there is an option to keep stdin open, I'm no expert either but this is good enough for me. Thank you for the answer, I appreciate it.

**Comment by Earlopain** [comment]
I just stumbled upon an option in compose I can use to remove this workaround: `tty: true`. Maybe someone else will find this useful.

**Comment by andsens** [comment]
`esbuild </dev/zero` is probably a bit nicer than `sleep |...`

**Comment by evanw** [comment]
You should almost certainly be using the new `--watch=forever` option, released in [version 0.15.7](https://github.com/evanw/esbuild/releases/v0.15.7) and mentioned in [the documentation for watch mode](https://esbuild.github.io/api/#watch). In any case `esbuild </dev/zero` isn't a good idea because it needlessly burns CPU.
